cli/azd/resources/scaffold/templates/resources.bicept

```diff
@@ -558,64 +557,17 @@ module {{bicepName .Name}} 'br/public:avm/res/app/container-app:0.8.0' = {
     scaleMaxReplicas: 10
     secrets: {
       secureList:  union([
-        {{- if .DbCosmosMongo}}
-        {
-          name: 'mongodb-url'
-          identity:{{bicepName .Name}}Identity.outputs.resourceId
-          keyVaultUrl: cosmos.outputs.exportedSecrets['MONGODB-URL'].secretUri
-        }
-        {{- end}}
-        {{- if (and .DbPostgres (eq .DbPostgres.AuthType "PASSWORD")) }}
-        {
-          name: 'postgresql-password'
-          value: postgreSqlDatabasePassword
-        }
-        {
-          name: 'postgresql-db-url'
-          value: 'postgresql://${postgreSqlDatabaseUser}:${postgreSqlDatabasePassword}@${postgreServer.outputs.fqdn}:5432/${postgreSqlDatabaseName}'
-        }
-        {{- end}}
-        {{- if (and .DbMySql (eq .DbMySql.AuthType "PASSWORD")) }}
-        {
-          name: 'mysql-password'
-          value: mysqlDatabasePassword
-        }
-        {
-          name: 'mysql-db-url'
-          value: 'mysql://${mysqlDatabaseUser}:${mysqlDatabasePassword}@${mysqlServer.outputs.fqdn}:3306/${mysqlDatabaseName}'
-        }
-        {{- end}}
-        {{- if .DbRedis}}
-        {
-          name: 'redis-pass'
-          identity:{{bicepName .Name}}Identity.outputs.resourceId
-          keyVaultUrl: '${keyVault.outputs.uri}secrets/REDIS-PASSWORD'
-        }
-        {
-          name: 'redis-url'
-          identity:{{bicepName .Name}}Identity.outputs.resourceId
-          keyVaultUrl: '${keyVault.outputs.uri}secrets/REDIS-URL'
-        }
-        {{- end}}
-        {{- if (and .AzureEventHubs (eq .AzureEventHubs.AuthType "CONNECTION_STRING")) }}
+        {{- range .EnvironmentVariableInformation.ValueSecretDefinitions}}
         {
-          name: 'event-hubs-connection-string'
-          identity:{{bicepName .Name}}Identity.outputs.resourceId
-          keyVaultUrl: '${keyVault.outputs.uri}secrets/EVENT-HUBS-CONNECTION-STRING'
+          name: '{{.SecretName}}'
+          value: '{{.SecretValue}}'
         }
         {{- end}}
-        {{- if (and .AzureServiceBus (eq .AzureServiceBus.AuthType "CONNECTION_STRING")) }}
+        {{- range .EnvironmentVariableInformation.KeyVaultSecretDefinitions}}
         {
-          name: 'servicebus-connection-string'
-          identity:{{bicepName .Name}}Identity.outputs.resourceId
-          keyVaultUrl: '${keyVault.outputs.uri}secrets/SERVICEBUS-CONNECTION-STRING'
-        }
-        {{- end}}
-        {{- if (and .AzureStorageAccount (eq .AzureStorageAccount.AuthType "CONNECTION_STRING")) }}
-        {
-          name: 'storage-account-connection-string'
-          identity:{{bicepName .Name}}Identity.outputs.resourceId
-          keyVaultUrl: '${keyVault.outputs.uri}secrets/STORAGE-ACCOUNT-CONNECTION-STRING'
+          name: '{{.SecretName}}'
+          identity:{{bicepName $service.Name}}Identity.outputs.resourceId
+          keyVaultUrl: '{{.KeyVaultUrl}}'
```

Comment:
Here, the outer scope is `EnvironmentVariableInformation`, not infraSpec. So I added variable: `service`